File: Nx3CS/bin/test/src/EReg.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
public  class EReg : global::haxe.lang.HxObject {
	public    EReg(global::haxe.lang.EmptyObject empty){
		unchecked {
			#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
			{
			}
			
		}
		#line default
	}
	
	
	public    EReg(string r, string opt){
		unchecked {
			#line 34 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
			global::EReg.__hx_ctor__EReg(this, r, opt);
		}
		#line default
	}
	
	
	public static   void __hx_ctor__EReg(global::EReg __temp_me6, string r, string opt){
		unchecked {
			#line 35 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
			int opts = ((int) (global::haxe.lang.Runtime.toInt(((object) (global::System.Text.RegularExpressions.RegexOptions.CultureInvariant) ))) );
			{
				#line 36 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
				int _g1 = 0;
				#line 36 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
				int _g = opt.Length;
				#line 36 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
				while (( _g1 < _g )){
					#line 36 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
					int i = _g1++;
					{
						#line 37 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
						int _g2 = default(int);
						#line 37 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
						_g2 = ((int) (opt[i]) );
						#line 37 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
						switch (_g2){
							case 105:
							{
								#line 40 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
								opts |= ((int) (global::haxe.lang.Runtime.toInt(((object) (global::System.Text.RegularExpressions.RegexOptions.IgnoreCase) ))) );
								#line 40 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
								break;
							}
							
							
							case 103:
							{
								#line 42 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
								__temp_me6.isGlobal = true;
								#line 42 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
								break;
							}
							
							
							case 109:
							{
								#line 44 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
								opts |= ((int) (global::haxe.lang.Runtime.toInt(((object) (global::System.Text.RegularExpressions.RegexOptions.Multiline) ))) );
								#line 44 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
								break;
							}
							
							
							case 99:
							{
								#line 46 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
								opts |= ((int) (global::haxe.lang.Runtime.toInt(((object) (global::System.Text.RegularExpressions.RegexOptions.Compiled) ))) );
								#line 46 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
								break;
							}
							
							
						}
						
					}
					
				}
				
			}
			
			#line 50 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
			__temp_me6.regex = new global::System.Text.RegularExpressions.Regex(((string) (r) ), ((global::System.Text.RegularExpressions.RegexOptions) (((object) (opts) )) ));
		}
		#line default
	}
	
	
	public static  new object __hx_createEmpty(){
		unchecked {
			#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
			return new global::EReg(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
		}
		#line default
	}
	
	
	public static  new object __hx_create(global::Array arr){
		unchecked {
			#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
			return new global::EReg(global::haxe.lang.Runtime.toString(arr[0]), global::haxe.lang.Runtime.toString(arr[1]));
		}
		#line default
	}
	
	
	public  global::System.Text.RegularExpressions.Regex regex;
	
	public  bool isGlobal;
	
	public virtual   string replace(string s, string @by){
		unchecked {
			#line 104 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
			if (this.isGlobal) {
				#line 104 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
				return this.regex.Replace(((string) (s) ), ((string) (@by) ));
			}
			 else {
				#line 104 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
				return this.regex.Replace(((string) (s) ), ((string) (@by) ), ((int) (1) ));
			}
			
		}
		#line default
	}
	
	
	public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
		unchecked {
			#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
			switch (hash){
				case 1821933:
				{
					#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
					this.isGlobal = global::haxe.lang.Runtime.toBool(@value);
					#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
					return @value;
				}
				
				
				case 1723805383:
				{
					#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
					this.regex = ((global::System.Text.RegularExpressions.Regex) (@value) );
					#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
					return @value;
				}
				
				
				default:
				{
					#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
					return base.__hx_setField(field, hash, @value, handleProperties);
				}
				
			}
			
		}
		#line default
	}
	
	
	public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
		unchecked {
			#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
			switch (hash){
				case 724060212:
				{
					#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
					return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("replace") ), ((int) (724060212) ))) );
				}
				
				
				case 1821933:
				{
					#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
					return this.isGlobal;
				}
				
				
				case 1723805383:
				{
					#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
					return this.regex;
				}
				
				
				default:
				{
					#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
					return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
				}
				
			}
			
		}
		#line default
	}
	
	
	public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
		unchecked {
			#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
			switch (hash){
				case 724060212:
				{
					#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
					return this.replace(global::haxe.lang.Runtime.toString(dynargs[0]), global::haxe.lang.Runtime.toString(dynargs[1]));
				}
				
				
				default:
				{
					#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
					return base.__hx_invokeField(field, hash, dynargs);
				}
				
			}
			
		}
		#line default
	}
	
	
	public override   void __hx_getFields(global::Array<object> baseArr){
		unchecked {
			#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
			baseArr.push("isGlobal");
			#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
			baseArr.push("regex");
			#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
			{
				#line 27 "F:\\HaxeToolkit\\haxe\\std\\cs\\_std\\EReg.hx"
				base.__hx_getFields(baseArr);
			}
			
		}
		#line default
	}
	
	
}

File: Nx3Java/java/src/haxe/root/EReg.java
package haxe.root;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class EReg extends haxe.lang.HxObject
{
	public    EReg(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    EReg(java.lang.String r, java.lang.String opt)
	{
		haxe.root.EReg.__hx_ctor__EReg(this, r, opt);
	}
	
	
	public static   void __hx_ctor__EReg(haxe.root.EReg __temp_me3, java.lang.String r, java.lang.String opt)
	{
		int flags = 0;
		{
			int _g1 = 0;
			int _g = opt.length();
			while (( _g1 < _g ))
			{
				int i = _g1++;
				{
					int _g2 = 0;
					if (( i < opt.length() )) 
					{
						_g2 = ((int) (opt.charAt(i)) );
					}
					 else 
					{
						_g2 = -1;
					}
					
					switch (_g2)
					{
						case 105:
						{
							flags |= java.util.regex.Pattern.CASE_INSENSITIVE;
							break;
						}
						
						
						case 109:
						{
							flags |= java.util.regex.Pattern.MULTILINE;
							break;
						}
						
						
						case 115:
						{
							flags |= java.util.regex.Pattern.DOTALL;
							break;
						}
						
						
						case 103:
						{
							__temp_me3.isGlobal = true;
							break;
						}
						
						
					}
					
				}
				
			}
			
		}
		
		__temp_me3.matcher = java.util.regex.Pattern.compile(haxe.lang.Runtime.toString(haxe.root.EReg.convert(r)), ((int) (flags) )).matcher(((java.lang.CharSequence) ("") ));
		__temp_me3.pattern = r;
	}
	
	
	public static   java.lang.String convert(java.lang.String r)
	{
		return r;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new haxe.root.EReg(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new haxe.root.EReg(haxe.lang.Runtime.toString(arr.__get(0)), haxe.lang.Runtime.toString(arr.__get(1)));
	}
	
	
	public  java.lang.String pattern;
	
	public  java.util.regex.Matcher matcher;
	
	public  java.lang.String cur;
	
	public  boolean isGlobal;
	
	public   boolean match(java.lang.String s)
	{
		this.cur = s;
		this.matcher = this.matcher.reset(((java.lang.CharSequence) (s) ));
		return this.matcher.find();
	}
	
	
	public   java.lang.String matched(int n)
	{
		if (( n == 0 )) 
		{
			return this.matcher.group();
		}
		 else 
		{
			return this.matcher.group(((int) (n) ));
		}
		
	}
	
	
	public   java.lang.String replace(java.lang.String s, java.lang.String by)
	{
		this.matcher.reset(((java.lang.CharSequence) (s) ));
		by = haxe.lang.StringExt.split(by, "$$").join("\\$");
		if (this.isGlobal) 
		{
			return this.matcher.replaceAll(haxe.lang.Runtime.toString(by));
		}
		 else 
		{
			return this.matcher.replaceFirst(haxe.lang.Runtime.toString(by));
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef260 = true;
			switch (field.hashCode())
			{
				case -568986259:
				{
					if (field.equals("isGlobal")) 
					{
						__temp_executeDef260 = false;
						this.isGlobal = haxe.lang.Runtime.toBool(value);
						return value;
					}
					
					break;
				}
				
				
				case -791090288:
				{
					if (field.equals("pattern")) 
					{
						__temp_executeDef260 = false;
						this.pattern = haxe.lang.Runtime.toString(value);
						return value;
					}
					
					break;
				}
				
				
				case 98880:
				{
					if (field.equals("cur")) 
					{
						__temp_executeDef260 = false;
						this.cur = haxe.lang.Runtime.toString(value);
						return value;
					}
					
					break;
				}
				
				
				case 840862002:
				{
					if (field.equals("matcher")) 
					{
						__temp_executeDef260 = false;
						this.matcher = ((java.util.regex.Matcher) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef260) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef261 = true;
			switch (field.hashCode())
			{
				case 1094496948:
				{
					if (field.equals("replace")) 
					{
						__temp_executeDef261 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("replace"))) );
					}
					
					break;
				}
				
				
				case -791090288:
				{
					if (field.equals("pattern")) 
					{
						__temp_executeDef261 = false;
						return this.pattern;
					}
					
					break;
				}
				
				
				case 840861988:
				{
					if (field.equals("matched")) 
					{
						__temp_executeDef261 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("matched"))) );
					}
					
					break;
				}
				
				
				case 840862002:
				{
					if (field.equals("matcher")) 
					{
						__temp_executeDef261 = false;
						return this.matcher;
					}
					
					break;
				}
				
				
				case 103668165:
				{
					if (field.equals("match")) 
					{
						__temp_executeDef261 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("match"))) );
					}
					
					break;
				}
				
				
				case 98880:
				{
					if (field.equals("cur")) 
					{
						__temp_executeDef261 = false;
						return this.cur;
					}
					
					break;
				}
				
				
				case -568986259:
				{
					if (field.equals("isGlobal")) 
					{
						__temp_executeDef261 = false;
						return this.isGlobal;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef261) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef262 = true;
			switch (field.hashCode())
			{
				case 1094496948:
				{
					if (field.equals("replace")) 
					{
						__temp_executeDef262 = false;
						return this.replace(haxe.lang.Runtime.toString(dynargs.__get(0)), haxe.lang.Runtime.toString(dynargs.__get(1)));
					}
					
					break;
				}
				
				
				case 103668165:
				{
					if (field.equals("match")) 
					{
						__temp_executeDef262 = false;
						return this.match(haxe.lang.Runtime.toString(dynargs.__get(0)));
					}
					
					break;
				}
				
				
				case 840861988:
				{
					if (field.equals("matched")) 
					{
						__temp_executeDef262 = false;
						return this.matched(((int) (haxe.lang.Runtime.toInt(dynargs.__get(0))) ));
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef262) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("isGlobal");
		baseArr.push("cur");
		baseArr.push("matcher");
		baseArr.push("pattern");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


